idaes/generic_models/unit_models/tests/test_skeleton_unit_model.py

```diff
@@ -0,0 +1,305 @@
+        m.fs.skeleton.pressure_out = Var(m.fs.time, initialize=101)
+
+        # Surrogate model equations
+        # Flow equation
+
+        def rule_flow(m, t, i):
+            return m.flow_comp_out[t, i] == m.flow_comp_in[t, i]
+
+        m.fs.skeleton.eq_flow = \
+            Constraint(m.fs.time, m.fs.skeleton.comp_list, rule=rule_flow)
+
+        m.fs.skeleton.eq_temperature = Constraint(
+            expr=m.fs.skeleton.temperature_out[0] ==
+            m.fs.skeleton.temperature_in[0] + 10)
+        m.fs.skeleton.eq_pressure = Constraint(
+            expr=m.fs.skeleton.pressure_out[0] ==
+            m.fs.skeleton.pressure_in[0] - 10)
+
+        inlet_dict = {"flow_mol_comp": m.fs.skeleton.flow_comp_in,
+                      "temperature": m.fs.skeleton.temperature_in,
+                      "pressure": m.fs.skeleton.pressure_in}
+        outlet_dict = {"flow_mol_comp": m.fs.skeleton.flow_comp_out,
+                       "temperature": m.fs.skeleton.temperature_out,
+                       "pressure": m.fs.skeleton.pressure_out}
+
+        m.fs.skeleton.add_ports(name="inlet", member_dict=inlet_dict)
+        m.fs.skeleton.add_ports(name="outlet", member_dict=outlet_dict)
+
+        return m
+
+    @pytest.mark.unit
+    def test_ports(self, skeleton_default):
+        # Check inlet port and port members
+        assert hasattr(skeleton_default.fs.skeleton, "inlet")
+        assert len(skeleton_default.fs.skeleton.inlet.vars) == 3
+        assert hasattr(skeleton_default.fs.skeleton.inlet, "flow_mol_comp")
+        assert hasattr(skeleton_default.fs.skeleton.inlet, "temperature")
+        assert hasattr(skeleton_default.fs.skeleton.inlet, "pressure")
+
+        # Check outlet port and port members
+        assert hasattr(skeleton_default.fs.skeleton, "outlet")
+        assert len(skeleton_default.fs.skeleton.outlet.vars) == 3
+        assert hasattr(skeleton_default.fs.skeleton.outlet, "flow_mol_comp")
+        assert hasattr(skeleton_default.fs.skeleton.outlet, "temperature")
+        assert hasattr(skeleton_default.fs.skeleton.outlet, "pressure")
+
+    @pytest.mark.unit
+    def test_build(self, skeleton_default):
+        # Check build of model
+        assert hasattr(skeleton_default.fs.skeleton, "flow_comp_in")
+        assert hasattr(skeleton_default.fs.skeleton, "temperature_in")
+        assert hasattr(skeleton_default.fs.skeleton, "pressure_in")
+
+        assert hasattr(skeleton_default.fs.skeleton, "eq_flow")
+
+        assert hasattr(skeleton_default.fs.skeleton, "flow_comp_out")
+        assert hasattr(skeleton_default.fs.skeleton, "temperature_out")
+        assert hasattr(skeleton_default.fs.skeleton, "pressure_out")
+
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_default_initialize(self, skeleton_default):
+        # Check default initialize method raises ConfigurationError
+
+        with pytest.raises(ConfigurationError):
```

Comment:
It is good practice to check the error message here to make sure you are not getting another unexpected error.

```
with pytest.raises(ConfigurationError, match="expected string"):
```